### PR TITLE
.Net: Migrate AzureOpenAITextToAudioService to Azure.AI.OpenAI SDK v2

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
This PR migrates AzureOpenAITextToAudioService to Azure.AI.OpenAI SDK v2.

### Description
1. The new `AzureOpenAITextToAudioExecutionSettings` class is added to represent prompt execution settings for `AzureOpenAITextToAudioService`. Both `AzureOpenAITextToAudioService` classes that SK has today use the same prompt execution settings class - [OpenAITextToAudioExecutionSettings](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.OpenAI/TextToAudio/OpenAITextToAudioExecutionSettings.cs). This is a breaking change that is tracked in the issue - https://github.com/microsoft/semantic-kernel/issues/7053, and it will be decided later whether to proceed with the change or roll it back.
2. The `ClientCore.TextToAudio.cs` class is refactored to use the new `AzureOpenAITextToAudioExecutionSettings` class.  
3. The `ClientCore.TextToAudio.cs` class is refactored to decide which model id to use - the one from the prompt execution setting, the one supplied when registering the connector, or to use the deployment name if no model id is provided. This is done for backward compatibility with the existing `AzureOpenAITextToAudioService`. https://github.com/microsoft/semantic-kernel/issues/7104
4. Service collection and kernel builder extension methods are added to register the service in the DI container.  
5. Unit and integration tests are added as well.

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        if (executionSettings is AzureOpenAITextToAudioExecutionSettings settings)
+        {
+            return settings;
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var azureOpenAIExecutionSettings = JsonSerializer.Deserialize<AzureOpenAITextToAudioExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:111]: If someone has existing code that uses `OpenAITextToAudioExecutionSettings` then this serialize/derserialize operation should convert the settings. Is that correct?

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            this.ThrowIfFrozen();
+            this._voice = value;
+        }
+    }
+
+    /// <summary>
+    /// The format to audio in. Supported formats are mp3, opus, aac, and flac.
+    /// </summary>
+    [JsonPropertyName("response_format")]
+    public string ResponseFormat
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:37]: This should be optional

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            this.ThrowIfFrozen();
+            this._responseFormat = value;
+        }
+    }
+
+    /// <summary>
+    /// The speed of the generated audio. Select a value from 0.25 to 4.0. 1.0 is the default.
+    /// </summary>
+    [JsonPropertyName("speed")]
+    public float Speed
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:52]: This should be optional

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 1 above)
Comment (westey-m) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:111]: Should we wrap this in a try catch block and also throw an exception with the same message as below, i.e. `Invalid execution settings, cannot convert...`, with whatever Deserialize throw as an inner exception?
Callers may not necessarily expect that we are trying to do json deserialization here, and multiple exception types for a similar error is harder to catch and handle.

**Review by westey-m — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceCollectionExtensionsTests.cs [hunk 4]
```diff
@@ -88,6 +89,25 @@ public void ServiceCollectionAddAzureOpenAITextEmbeddingGenerationAddsValidServi
...
+        // Arrange
+        var sut = new ServiceCollection();
+
+        // Act
+        var service = sut.AddAzureOpenAITextToAudio("deployment-name", "https://endpoint", "api-key")
+            .BuildServiceProvider()
+            .GetRequiredService<ITextToAudioService>();
+
+        // Assert
+        Assert.IsType<AzureOpenAITextToAudioService>(service);
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceCollectionExtensionsTests.cs:106]: Not needed.

```suggestion
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceKernelBuilderExtensionsTests.cs [hunk 5]
```diff
@@ -88,6 +89,25 @@ public void KernelBuilderAddAzureOpenAITextEmbeddingGenerationAddsValidService(I
...
+        // Arrange
+        var sut = Kernel.CreateBuilder();
+
+        // Act
+        var service = sut.AddAzureOpenAITextToAudio("deployment-name", "https://endpoint", "api-key")
+            .Build()
+            .GetRequiredService<ITextToAudioService>();
+
+        // Assert
+        Assert.IsType<AzureOpenAITextToAudioService>(service);
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceKernelBuilderExtensionsTests.cs:106]: Not needed

```suggestion
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Services/AzureOpenAITextToAudioServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        var service = new AzureOpenAITextToAudioService("deployment-name", "https://endpoint", "api-key", "model-id", this._httpClient);
+
+        // Act & Assert
+        await Assert.ThrowsAsync<NotSupportedException>(async () => await service.GetAudioContentsAsync("Some text", new AzureOpenAITextToAudioExecutionSettings() { ResponseFormat = "not supported" }));
+    }
+
+    [Theory]
+    [InlineData(true, "http://local-endpoint")]
+    [InlineData(false, "https://endpoint")]
+    public async Task GetAudioContentUsesValidBaseUrlAsync(bool useHttpClientBaseAddress, string expectedBaseAddress)
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Services/AzureOpenAITextToAudioServiceTests.cs:158]: Add examples with path: `https://endpoint/path/`

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.TextToAudio.cs [hunk 7]
```diff
@@ -64,4 +70,12 @@ private static (GeneratedSpeechFormat Format, string MimeType) GetGeneratedSpeec
             "PCM" => (GeneratedSpeechFormat.Pcm, "audio/l16"),
             _ => throw new NotSupportedException($"The format '{format}' is not supported.")
         };
+
+    private string GetModelId(AzureOpenAITextToAudioExecutionSettings executionSettings, string? modelId)
+    {
+        return
+            !string.IsNullOrWhiteSpace(modelId) ? modelId! :
+            !string.IsNullOrWhiteSpace(executionSettings.ModelId) ? executionSettings.ModelId! :
+            this.DeploymentOrModelName;
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.TextToAudio.cs:79]: On OpenAI V2 I'm using ModelId, not name, suggest making this update too here or later.

```suggestion
            this.DeploymentOrModelId;
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Extensions/AzureOpenAIServiceCollectionExtensions.cs [hunk 8]
```diff
@@ -234,6 +235,46 @@ public static class AzureOpenAIServiceCollectionExtensions
...
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> instance to augment.</param>
+    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="modelId">Model identifier, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="services"/>.</returns>
+    [Experimental("SKEXP0001")]
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Extensions/AzureOpenAIServiceCollectionExtensions.cs:251]: This is an experimental feature of the Connector not Abstractions, this is a bug, might raise breaking changes.

Same for others.

```suggestion
    [Experimental("SKEXP0010")]
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureOpenAI;
+
+/// <summary>
+/// Execution settings for Azure OpenAI text-to-audio request.
+/// </summary>
+[Experimental("SKEXP0001")]
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:15]:
```suggestion
[Experimental("SKEXP0010")]
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+public sealed class AzureOpenAITextToAudioExecutionSettings : PromptExecutionSettings
+{
+    /// <summary>
+    /// The voice to use when generating the audio. Supported voices are alloy, echo, fable, onyx, nova, and shimmer.
+    /// </summary>
+    [JsonPropertyName("voice")]
+    public string Voice
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:22]: Agree of Mark, optional for this too, and use it as a property, not sure whi this is a field, same for others.

```suggestion
    public string Voice { get; } = DefaultVoice;
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var azureOpenAIExecutionSettings = JsonSerializer.Deserialize<AzureOpenAITextToAudioExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
+
+        if (azureOpenAIExecutionSettings is not null)
+        {
+            return azureOpenAIExecutionSettings;
+        }
```
Comment (RogerBarreto) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:116]: We may not need to check null here.

```suggestion
        return azureOpenAIExecutionSettings!;
```

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.TextToAudio.cs (hunk 7 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.TextToAudio.cs:79]: Yep, I'll do it in a separate PR later.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 10 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:22]: We'll update all properties of all new *PromptExecutionSettings classes in on go in scope of the issue - https://github.com/microsoft/semantic-kernel/issues/7111.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 3 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:52]: We'll update all properties of all new *PromptExecutionSettings classes in on go in scope of the issue - https://github.com/microsoft/semantic-kernel/issues/7111.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:37]: We'll update all properties of all new *PromptExecutionSettings classes in on go in scope of the issue - https://github.com/microsoft/semantic-kernel/issues/7111.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 1 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:111]: Yes, correct, and as discussed offline the code should work. Anyway, we'll doble check it in scope of this issue - https://github.com/microsoft/semantic-kernel/issues/7110

Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:111]: Thanks, that makes sense. I'll do it in the next clean-up round sometime later for all new prompt execution settings classes in one go.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 9 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs]: Fixed.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Extensions/AzureOpenAIServiceCollectionExtensions.cs (hunk 8 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Extensions/AzureOpenAIServiceCollectionExtensions.cs]: Fixed.

File: dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Services/AzureOpenAITextToAudioServiceTests.cs (hunk 6 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Services/AzureOpenAITextToAudioServiceTests.cs:158]: Done!

File: dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceCollectionExtensionsTests.cs (hunk 4 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceCollectionExtensionsTests.cs:107]: Discussed offline and agreed to keep it.

File: dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceKernelBuilderExtensionsTests.cs (hunk 5 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI.UnitTests/Extensions/AzureOpenAIServiceKernelBuilderExtensionsTests.cs]: Discussed offline and agreed to keep it.

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs (hunk 11 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs:116]: As discussed offline, all the new prompt execution settings will be refactored in one go later, within the scope of a separate PR.

**Review by RogerBarreto — APPROVED** [review]

**Pull request merged**